### DISCUSSION TITLE
Support adding documentation as context for AI completions

### DISCUSSION THREAD

**Discussion by evelant** [zed-industries/zed#37856] [Feature Requests]
### Check for existing issues

- [X] Completed

### Describe the feature

A killer feature of https://cursor.sh is that it allows indexing of documentation websites to improve AI responses. This is hugely useful as much of the time without docs AI code completion will use outdated or incorrect APIs of many libraries. It would be awesome if Zed could do this as well!

**Comment by iamnbutler** [comment]
There is a bunch of code for bringing context into completions-now that we are open source this would be a great thing for someone to pick up and build ☺️

**Comment by alexrkaufman** [comment]
Related request: #18903

**Comment by fourcolors** [comment]
There could be an extension with MCP https://github.com/hannesrudolph/mcp-ragdocs i'd prefer updating that repo to support more RAG + Models
